bouncer/src/repo/core/model/bson/repo_bson_factory.cpp

```diff
@@ -25,6 +25,203 @@
 
 using namespace repo::core::model;
 
+/**
+* Specialisation of the BSON Builder dedicated to building Meshes and
+* Supermeshes. This is for internal use by the RepoBSONFactory and is
+* intended to build immutable nodes via one of the makeMeshNode
+* overloads.
+*/
+class RepoMeshBSONBuilder : public RepoBSONBuilder
+{
+	using BinMapping = std::unordered_map<std::string, std::pair<std::string, std::vector<uint8_t>>>;
+public:
+	std::string uniqueID;
+
+	BinMapping& mapping()
+	{
+		return binMapping;
+	}
+
+	size_t& bytesize() {
+		return _byteSize;
+	}
+
+	/*
+	* TODO: because mongo has a stupid internal limit of 64MB, we can't store everything in a BSON
+	* There are 2 options
+	* 1. store binaries in memory outside of the bson and put it into GRIDFS at the point of commit
+	* 2. leave mongo's bson, use our own/exteral library that doesn't have this limit and database handle this at the point of commit
+	* below uses option 1, but ideally we should be doing option 2.
+	*/
+
+	template<typename T>
+	void appendLargeArray(std::string name, std::vector<T> data)
+	{
+		uint64_t byteCount = data.size() * sizeof(data[0]);
+
+		if (byteCount + _byteSize >= REPO_BSON_MAX_BYTE_SIZE) //inclusion of this binary exceeds the maximum, store separately
```

Comment:
if we're changing this anyway... 😆 

I believe we can just push it into binMapping regardless.. As we never store it as part of mongobson anymore (we always flush it out just before committing)

Reply:
Done!